external-test-clients/src/main/java/io/enmasse/test/ProbeClient.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2019, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.test;
+
+import io.enmasse.address.model.Address;
+import io.enmasse.address.model.AddressBuilder;
+import io.enmasse.address.model.AddressList;
+import io.enmasse.address.model.AddressSpace;
+import io.enmasse.address.model.AddressSpaceList;
+import io.enmasse.address.model.CoreCrd;
+import io.enmasse.address.model.DoneableAddress;
+import io.enmasse.address.model.DoneableAddressSpace;
+import io.enmasse.address.model.EndpointStatus;
+import io.enmasse.metrics.api.MetricType;
+import io.enmasse.metrics.api.MetricValue;
+import io.enmasse.metrics.api.Metrics;
+import io.enmasse.metrics.api.MetricsFormatter;
+import io.enmasse.metrics.api.ScalarMetric;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.NamespacedKubernetesClient;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.proton.ProtonClient;
+import io.vertx.proton.ProtonClientOptions;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonReceiver;
+import io.vertx.proton.ProtonSender;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.amqp.messaging.AmqpValue;
+import org.apache.qpid.proton.message.Message;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ProbeClient extends AbstractVerticle {
+    private final String host;
+    private final int port;
+    private final String address;
+
+    private ProbeClient(String host, int port, String address) {
+        this.host = host;
+        this.port = port;
+        this.address = address;
+    }
+
+    @Override
+    public void start(Future<Void> startPromise) {
+        ProtonClientOptions protonClientOptions = new ProtonClientOptions()
+                .setSsl(true)
+                .setTrustAll(true)
+                .setHostnameVerificationAlgorithm("");
+
+        ProtonClient client = ProtonClient.create(vertx);
+        client.connect(protonClientOptions, host, port, connectResult -> {
+            if (connectResult.succeeded()) {
+                ProtonConnection connection = connectResult.result();
+                ProtonReceiver receiver = connection.createReceiver(address);
+                receiver.handler((protonDelivery, message) -> {
+                    startPromise.complete();
```

Comment:
Organise to fail the promise if the message isn't received in a reasonable time period.